parsons/empower/empower.py

```diff
@@ -0,0 +1,261 @@
+        """
+        Gets fresh data from Empower API based on cache setting.
+        """
+
+        if not cache or self.data is None:
+            r = self.client.get_request(self.empower_uri)
+            logger.info("Empower data downloaded.")
+            return r
+
+        else:
+            return self.data
+
+    def _unix_convert(self, ts):
+        """
+        Converts UNIX timestamps to readable timestamps.
+        """
+
+        ts = datetime.utcfromtimestamp(int(ts) / 1000)
+        ts = ts.strftime("%Y-%m-%d %H:%M:%S UTC")
+        return ts
+
+    def _empty_obj(self, obj_name):
+        """
+        Determine if a dict object is empty.
+        """
+
+        if len(self.data[obj_name]) == 0:
+            return True
+        else:
+            return False
+
+    def get_profiles(self):
+        """
+        Get Empower profiles.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["profiles"])
+        for col in ["createdMts", "lastUsedEmpowerMts", "updatedMts"]:
+            tbl.convert_column(col, lambda x: self._unix_convert(x))
+        tbl.remove_column("activeCtaIds")  # Get as a method via get_profiles_active_ctas
+        return tbl
+
+    def get_profiles_active_ctas(self):
+        """
+        Get active ctas assigned to Empower profiles.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["profiles"]).long_table("eid", "activeCtaIds")
+        return tbl
+
+    def get_regions(self):
+        """
+        Get Empower regions.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["regions"])
+        tbl.convert_column("inviteCodeCreatedMts", lambda x: self._unix_convert(x))
+        return tbl
+
+    def get_cta_results(self):
+        """
+        Get Empower call to action results.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["ctaResults"])
+        tbl.convert_column("contactedMts", lambda x: self._unix_convert(x))
+        tbl = tbl.unpack_nested_columns_as_rows(
+            "answerIdsByPromptId", key="profileEid", expand_original=True
+        )
+        tbl.unpack_list("answerIdsByPromptId_value", replace=True)
+        col_list = [v for v in tbl.columns if v.find("value") != -1]
+        tbl.coalesce_columns("answer_id", col_list, remove_source_columns=True)
+        tbl.remove_column("uid")
+        tbl.remove_column("answers")  # Per docs, this is deprecated.
+        return tbl
+
+    def _split_ctas(self):
+        """
+        Internal method to split CTA objects into tables.
+        """
+
+        ctas = Table(self.data["ctas"])
+        for col in [
```

Comment:
formatting looks weird here but if it passes our linter/formatter I guess that's fine